packages/jsii/lib/assembler.ts

```diff
@@ -3102,17 +3102,42 @@ function noEmptyDict<T>(
 function toDependencyClosure(assemblies: readonly spec.Assembly[]): {
   [name: string]: spec.AssemblyConfiguration;
 } {
-  const result: { [name: string]: spec.AssemblyTargets } = {};
+  const result: { [name: string]: spec.AssemblyConfiguration } = {};
   for (const assembly of assemblies) {
     if (!assembly.targets) {
       continue;
     }
     result[assembly.name] = {
-      submodules: assembly.submodules,
+      submodules: cleanUp(assembly.submodules),
       targets: assembly.targets,
     };
   }
   return result;
+
+  /**
+   * Removes unneeded fields from the entries part of the `dependencyClosure`
+   * property. Fields such as `readme` are not necessary and can bloat up the
+   * assembly object.
+   *
+   * This removes the `readme` and `locationInModule` fields from the submodule
+   * descriptios if present.
+   *
+   * @param submodules the submodules list to clean up.
+   *
+   * @returns the cleaned up submodules list.
+   */
+  function cleanUp(
```

Comment:
This should be accompanied by a different type signature in the spec, no? 

Those fields should not even be present in a dependency's submodules field.